verticapy/pipeline/_train.py

```diff
@@ -0,0 +1,222 @@
+
+    Parameters
+    ----------
+    train: dict
+        YAML object which outlines the steps of the operation.
+    vdf: vDataFrame
+        The model trained in the training step.
+    pipeline_name: str
+        The prefix name of the intended pipeline to unify
+        the creatation of the objects.
+    cols: SQLColumns
+        ``list`` of the columns needed to deploy the model.
+
+    Returns
+    -------
+    return meta_sql, model, model_sql
+    str
+        The SQL to replicate the steps of the yaml file.
+    VerticaModel
+        The model created after training.
+    str
+        The SQL needed to retrain the model.
+    """
+    meta_sql = ""
+    if "train_test_split" in train:
+        info = train["train_test_split"]
+        test_size = 0.33 if "test_size" not in info else info["test_size"]
+        train_set, test_set = vdf.train_test_split(test_size)
+        meta_sql += execute_and_add(
+            f"CREATE OR REPLACE VIEW {pipeline_name + '_TRAIN_VIEW'} AS SELECT * FROM "
+            + train_set.current_relation()
+            + ";"
+        )
+        meta_sql += execute_and_add(
+            f"CREATE OR REPLACE VIEW {pipeline_name + '_TEST_VIEW'} AS SELECT * FROM "
+            + test_set.current_relation()
+            + ";"
+        )
+
+    else:
+        meta_sql += execute_and_add(
+            f"CREATE OR REPLACE VIEW {pipeline_name + '_TRAIN_VIEW'} AS SELECT * FROM "
+            + vdf.current_relation()
+            + ";"
+        )
+        meta_sql += execute_and_add(
+            f"CREATE OR REPLACE VIEW {pipeline_name + '_TEST_VIEW'} AS SELECT * FROM "
+            + vdf.current_relation()
+            + ";"
+        )
+
+    methods = list(train.keys())
+    methods = [method for method in methods if "method" in method]
+    for method in methods:
+        tf = train[method]
+        temp_str = ""
+        name = tf["name"]
+        target = tf["target"] if "target" in tf else ""
+        params = tf["params"] if "params" in tf else []
+        temp_str = ""
+        for param in params:
+            temp = params[param]
+            if isinstance(temp, str):
+                temp_str += f"{param} = '{params[param]}', "
+            else:
+                temp_str += f"{param} = {params[param]}, "
+        temp_str = temp_str[:-2]
+        model = None
+        eval(
+            f"exec(\"model = {name}('{pipeline_name + '_MODEL'}', {temp_str})\")",
+            globals(),
+        )
+        predictors = cols  # ['"col1"', '"col2"', '"col3"', '"col4"']
+        if "include" in tf:
+            predictors = tf["include"]
+        if "exclude" in tf:
+            predictors = list(set(predictors) - set(tf["exclude"]))
+        if target == "":
+            # UNSUPERVISED
+            model.fit(pipeline_name + "_TRAIN_VIEW", predictors)
+        else:
+            # SUPERVISED
+            model.fit(pipeline_name + "_TRAIN_VIEW", predictors, target)
+        meta_sql += "\n"
+
+        try:
```

Comment:
In general it is not very good practice to use try/catch for code flow. In this case you are relying on the QueryError being due to an unsupervised model, when it might be due to some other issue. Is there any way to handle this with an if/else?